pkg/controller/nstemplateset/cluster_resources_test.go

```diff
@@ -370,13 +371,14 @@ func TestPromoteClusterResources(t *testing.T) {
 				HasConditions(Provisioning())
 			AssertThatCluster(t, cl).
 				HasResource("for-"+username, &quotav1.ClusterResourceQuota{},
+					WithLabel("toolchain.dev.openshift.com/templateref", "advanced-cluster-12345bb"),
 					WithLabel("toolchain.dev.openshift.com/tier", "advanced"),
 					Containing(`"limits.cpu":"2","limits.memory":"10Gi"`)) // upgraded
 		})
 
 		t.Run("no redundant cluster resource quota to be deleted for the given user", func(t *testing.T) {
 			// given
-			nsTmplSet := newNSTmplSet(namespaceName, username, "advanced", withConditions(Provisioned())) // no cluster resources, so the "advancedCRQ" should be deleted
+			nsTmplSet := newNSTmplSet(namespaceName, username, "advanced", withConditions(Provisioned()), withClusterResources())
```

Comment:
you're changing the initial state of the test here. In this case, we are supposed to deal with an `NSTemplateSet` in which no `clusterResources` are defined.  So should we add a test where there is no `clusterResources` at all, or are we going to drop this case?

Reply:
well, the original test was wrong - look at the flow - it doesn't expect that anything should be deleted, but there is a cluster resource owned by the user and with `advanced` so it should actually expect that the resources should be deleted.

> So should we add a test where there is no clusterResources at all, or are we going to drop this case?

There are is a test where  `clusterResources` template is  not set, but I can add one more for this specific case

Reply:
added here: eba87b7